sirepo/simulation_db.py

```diff
@@ -1000,7 +978,9 @@ def _pkg_relative_path_static(file_dir, file_name):
     Returns:
         str: full relative path of the file
     """
-    return '/' + RESOURCE_FOLDER.bestrelpath(static_file_path(file_dir, file_name))
+    # POSIT: static files outside of Sirepo use the same path structure relative
+    # to the static dir
+    return '/' + sirepo.resource.static(file_dir, file_name, relpath=True)
```

Comment:
This is an example where we need to think about adding a parameter. There's a lot going on here that's subtle. First, it's a URL that's being returned, not a "relpath", which implies a file. Second, resource.static is about files, not URLs. Third, this is in the context of a schema initialization. It's called in one place to do a very specific thing. The caller of this routine is very complicated. Too complicated imiho.

That's why I think relpath should be removed from pkresource and sirepo.resource, and this specific thing is to be handled in the routine here, or rather, the function itself should be unwound a bit. I can help with this. 

We should definitely have a test for compiling schemas to expected results.

Reply:
bc12ff9ff